smarts/core/lanepoints.py

```diff
@@ -921,6 +921,18 @@ def paths_starting_at_lanepoint(
         Returns:
             All branches(as lists) stemming from the lanepoint.
         """
+        # Early exit if there are no valid paths ahead in the route for this lane
```

Comment:
What should the result be if there are no valid lanepoints that would be returned? It is only non-empty right now because we always append the lanepoint that was passed in (even if it is not on the route, which is where the bug came from).

Reply:
If there are no valid lane points then an empty list. I am just noticing lane points that are on the whitelisted roads are being excluded. The current interface is lying by the omission of the fact that lane points on lanes which do not exit onto a whitelisted road will also be excluded.

Reply:
We discussed this offline, and I have changed the variable name to `route_edge_ids` and added more detail to the docstring about the function's behaviour.